### PR TITLE
lsp: implement used refs completions

### PR DESCRIPTION
**Pull request by charlieegan3**
This implements a completions provider that makes suggestions based on refs users have already typed into their files.

Fixes https://github.com/StyraInc/regal/issues/774

### CODE REVIEW

File: internal/lsp/completions/refs/used.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+					"capabilities": rio.ToMap(config.CapabilitiesForThisVersion()),
+				},
+			},
+		},
+	}
+
+	regoArgs := []func(*rego.Rego){
+		rego.ParsedBundle("regal", &regalRules),
+		rego.Module("module.rego", `
+package lsp.completions
```
Comment (anderseknert) [inline internal/lsp/completions/refs/used.go:44]: I'd much prefer it if this was a .rego file embedded in the binary. That way we can work with it as any other Rego... lint, test, so on. Was there anything that prevented that?

**Review by anderseknert — APPROVED** [review]
Looks great! And I'm excited to try it out 😃 Just one comment / request, but it's not blocking.

**Review by srenatus — APPROVED** [review]
Nicely done. 👍

File: internal/lsp/completions/refs/used.go (hunk 1 above)
Comment (charlieegan3) [reply internal/lsp/completions/refs/used.go]: Yeah I should do that. It was only 3 lines or something at the start but it got more involved as I discovered some edge cases... 

Fixed in 88e76666c426da73f737aa4fa00750761d3614bb

**Review by srenatus — APPROVED** [review]

File: internal/lsp/completions/refs/used.go (hunk 1 above)
Comment (anderseknert) [reply internal/lsp/completions/refs/used.go]: Nice, thanks!

**Pull request merged**